common/reactive/src/main/java/io/helidon/common/reactive/IoMulti.java

```diff
@@ -67,23 +103,175 @@ static OutputStreamMultiBuilder builderOutputStream() {
     static Multi<ByteBuffer> createInputStream(final InputStream inputStream) {
         return IoMulti.builderInputStream(inputStream)
                 .build();
     }
 
+    /**
+     * Create a {@link Multi} instance that publishes {@link ByteBuffer}s from
+     * the given {@link InputStream}.
+     * <p>
+     * {@link InputStream} is trusted not to block on read operations, in case
+     * it can't be assured use builder to specify executor for asynchronous waiting
+     * for blocking reads. {@code IoMulti.builder(is).executor(executorService).build()}.
+     *
+     * @param inputStream the Stream to publish
+     * @return Multi
+     */
+    static Multi<ByteBuffer> multiFromStream(final InputStream inputStream) {
+        return IoMulti.builderInputStream(inputStream)
+                .build();
+    }
+
     /**
      * Creates a builder of the {@link Multi} from supplied {@link java.io.InputStream}.
      *
      * @param inputStream the Stream to publish
      * @return the builder
+     * @deprecated Please use {@link #multiFromStreamBuilder(java.io.InputStream)}
      */
+    @Deprecated(since = "2.0.0", forRemoval = true)
     static MultiFromInputStreamBuilder builderInputStream(final InputStream inputStream) {
         Objects.requireNonNull(inputStream);
         return new MultiFromInputStreamBuilder(inputStream);
     }
 
+    /**
+     * Creates a builder of the {@link Multi} from supplied {@link java.io.InputStream}.
+     *
+     * @param inputStream the Stream to publish
+     * @return the builder
+     */
+    static MultiFromInputStreamBuilder multiFromStreamBuilder(final InputStream inputStream) {
+        Objects.requireNonNull(inputStream);
+        return new MultiFromInputStreamBuilder(inputStream);
+    }
+
+    /**
+     * Creates a multi that reads data from the provided byte channel.
+     * The multi uses an executor service to process asynchronous reads.
+     * You can provide a custom executor service using
+     * {@link #multiFromByteChannelBuilder(java.nio.channels.ReadableByteChannel)}.
+     *
+     * @param byteChannel readable byte channel with data
+     * @return publisher of data from the provided channel
+     */
+    static Multi<ByteBuffer> multiFromByteChannel(ReadableByteChannel byteChannel) {
+        return multiFromByteChannelBuilder(byteChannel).build();
+    }
+
+    /**
+     * Creates a builder of {@link Multi} from provided {@link java.nio.channels.ReadableByteChannel}.
+     *
+     * @param byteChannel readable byte channel with data
+     * @return fluent API builder to configure additional details
+     */
+    static MultiFromByteChannelBuilder multiFromByteChannelBuilder(ReadableByteChannel byteChannel) {
+        return new MultiFromByteChannelBuilder(Objects.requireNonNull(byteChannel));
+    }
+
+    /**
+     * Fluent API builder for creating a {@link io.helidon.common.reactive.Multi} from an
+     * {@link java.io.InputStream}.
```

Comment:
{@link ReadableByteChannel}.

Reply:
Not sure I understand the comment? 
In general it is preferable to use fully qualified class names in Javadoc, so you do not import a class only referenced in Javadoc

Reply:
Its just its on `MultiFromByteChannelBuilder` so it should be from `ReadableByteChannel` not from `InputStream`

Reply:
Fixed.